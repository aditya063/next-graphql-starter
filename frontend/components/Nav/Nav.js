import Link from 'next/link';
import styled from 'styled-components';

const StyledNav = styled.nav`
  a {
    color: white;
    letter-spacing: 1px;
    text-decoration: none;
    background: rgba(0, 0, 0, 0.2);
    display: flex;
    height: 100%;
    align-items: center;
    justify-content: center;
    transition: all 0.5s;
  }

  a:hover {
    background: rgba(0, 0, 0, 0.3);
  }

  ul {
    display: flex;
    padding: 0;
    margin: 0;
    list-style: none;
    height: 100%;
    flex-wrap: wrap;
  }

  li {
    flex: 3;
    flex: 1 1 50%;
  }

  @media all and (max-width: 500px) {
    li {
      flex-basis: 100%;
    }
  }
`;

class Nav extends React.PureComponent {
  render() {
    return (
      <StyledNav>
        <ul>
          <li>
            <Link href="/">
              <a>Home</a>
            </Link>
          </li>
          <li>
            <Link href="examples">
              <a>Examples</a>
            </Link>
          </li>
        </ul>
      </StyledNav>
    );
  }
}

export default Nav;
