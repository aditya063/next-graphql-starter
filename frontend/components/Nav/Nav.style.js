import styled from 'styled-components';

export const ul = styled.ul`
  display: flex;
  padding: 0;
  margin: 0;
  list-style: none;
  height: 100%;
  flex-wrap: wrap;
`;

export const li = styled.li`
  flex: 3;
  flex: 1 1 50%;

  @media all and (max-width: 500px) {
    flex-basis: 100%;
  }
`;

export const a = styled.a`
  color: white;
  letter-spacing: 1px;
  background: rgba(0, 0, 0, 0.2);
  display: flex;
  height: 100%;
  align-items: center;
  justify-content: center;
  transition: all 0.5s;
  cursor: pointer;

  :hover {
    background: rgba(0, 0, 0, 0.3);
  }
`;
