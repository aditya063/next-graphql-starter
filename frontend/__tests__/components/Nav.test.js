import React from 'react';
import { shallow } from 'enzyme';
import Nav from '../../components/Nav/Nav';

describe('Nav', () => {
  let mockProps;
  let wrapper;

  beforeEach(() => {
    jest.resetAllMocks();
    mockProps = {};
    wrapper = shallow(<Nav {...mockProps} />, {
      disableLifecycleMethods: true
    });
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
