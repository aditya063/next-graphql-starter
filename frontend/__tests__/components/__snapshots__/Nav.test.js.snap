// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav matches snapshot 1`] = `
<Nav__StyledNav>
  <ul>
    <li>
      <Link
        href="/"
      >
        <a>
          Home
        </a>
      </Link>
    </li>
    <li>
      <Link
        href="examples"
      >
        <a>
          Examples
        </a>
      </Link>
    </li>
  </ul>
</Nav__StyledNav>
`;
